### DISCUSSION TITLE
Error ICE31 TextStyle table is missing

### DISCUSSION THREAD

**Discussion by slide** [wixtoolset/issues#8879] [Questions]
I have been working on upgrading the installer for the Jenkins project from Wix3 to Wix5. Everything has been going swimingly except for an error I am now getting that I haven't been able to find any information about online.

I am getting several errors like the following:

` D:\a\wix\wix\src\ext\UI\wixlib\MaintenanceWelcomeDlg.wxs(17): error WIX0204: ICE31: Control MaintenanceWelcomeDlg.Title uses text style WixUI_Font_Bigger, but the TextStyle table does not exist. [C:\Users\slide\source\r 
epos\packaging-wix5\msi\build\Setup\Setup.wixproj]
  C:\Users\slide\source\repos\packaging-wix5\msi\build\Setup\JavaBrowseDlg.wxs(26): error WIX0204: ICE31: Control JavaBrowseDlg.Title uses text style WixUI_Font_Title, but the TextStyle table does not exist. [C:\Users\sli 
de\source\repos\packaging-wix5\msi\build\Setup\Setup.wixproj]
  D:\a\wix\wix\src\ext\UI\wixlib\DiskCostDlg.wxs(16): error WIX0204: ICE31: Control DiskCostDlg.Title uses text style WixUI_Font_Title, but the TextStyle table does not exist. [C:\Users\slide\source\repos\packaging-wix5\m 
si\build\Setup\Setup.wixproj]
  D:\a\wix\wix\src\ext\UI\wixlib\FatalError.wxs(15): error WIX0204: ICE31: Control FatalError.Title uses text style WixUI_Font_Bigger, but the TextStyle table does not exist. [C:\Users\slide\source\repos\packaging-wix5\ms 
i\build\Setup\Setup.wixproj]
  D:\a\wix\wix\src\ext\UI\wixlib\FilesInUse.wxs(22): error WIX0204: ICE31: Control FilesInUse.Title uses text style WixUI_Font_Title, but the TextStyle table does not exist. [C:\Users\slide\source\repos\packaging-wix5\msi 
\build\Setup\Setup.wixproj]
  D:\a\wix\wix\src\ext\UI\wixlib\MsiRMFilesInUse.wxs(30): error WIX0204: ICE31: Control MsiRMFilesInUse.Title uses text style WixUI_Font_Title, but the TextStyle table does not exist. [C:\Users\slide\source\repos\packagin 
g-wix5\msi\build\Setup\Setup.wixproj]`

Can someone point me to any resource that might help me resolve this issue? I can't figure out what could be causing it.

**Comment by bevanweiss** [comment]
Those validation errors say that the text styles being used by the UI haven't actually been specified in a TextStyle table within the finished MSI.
https://learn.microsoft.com/en-us/windows/win32/msi/textstyle-table

The various WixUI elements do have these defined, have you removed them from the WixUI templates that you're using?
https://github.com/wixtoolset/wix/blob/523c66a62a619e6aa9f30070173ea33edfb5e328/src/ext/UI/wixlib/WixUI_Mondo.wxs#L49-L50
